database/sqlite/sqlite_functions.c

```diff
@@ -905,6 +905,49 @@ int sql_store_dimension(
     return 1;
 }
 
+/*
+ * Store set option for a dimension
+ */
+int sql_set_dimension_option(uuid_t *dim_uuid, char *option)
+{
+    sqlite3_stmt *res = NULL;
+    int rc;
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE)
+            return 0;
+        error_report("Database has not been initialized");
+        return 1;
+    }
+
+    rc = sqlite3_prepare_v2(db_meta, "UPDATE dimension SET options = @options WHERE dim_id = @dim_id", -1, &res, 0);
```

Comment:
- How are we going to serialize/deserialize options if we need to save multiple of them in the future?
- Is storing the raw rrdim options a better/simpler approach? We are a long way from exhausting the available bits.

Reply:
Its good point but currently we don't have a use case I can think of that needs to persist any other option. If needed we can easily migrate in the future to store raw rrdim options